debridge_node/src/modules/chain/scanning/services/SolanaReaderService.ts

```diff
@@ -1,161 +1,127 @@
-          transaction.events.forEach(event => {
-            events.push({
-              ...event,
-              transactionHash: transaction.transactionHash,
-              slotNumber: transaction.slotNumber,
-            } as SolanaEvent);
-          });
-        });
-      this.logger.verbose(`Events ${events.length} from transaction are received`);
-
-      const submissions = events.map(event => {
-        try {
-          const submission = this.transformService.generateSubmissionFromSolanaEvent(event);
-          this.logger.verbose(`submission ${event.submissionId} is generated`);
-          return submission;
-        } catch (e) {
-          this.logger.error(`Can't generate submission ${event.submissionId} from solana event`);
-          this.logger.error(e);
-          throw e;
+    for await (const response of this.#duplex.responses) {
+      if (response.sendEventMessage?.oneofKind == undefined) {
+        continue;
+      }
+      switch (response.sendEventMessage.oneofKind) {
+        case undefined: {
+          continue;
         }
-      });
+        case 'heartbeat': {
+          // @ts-ignore
+          this.#logger.verbose(`Heartbeat: ${JSON.stringify(response.sendEventMessage.heartbeat)}`);
 
-      //sort in asc, we need it for correct updating last tracked value and nonce validation
-      submissions.sort((a, b) => a.nonce - b.nonce);
-      this.logger.verbose(`Events ${submissions.length} are prepared for db storing`);
-
-      if (submissions.length > 0) {
-        const result = await this.chainProcessingService.process(submissions, chainId, submissions.at(-1).txHash);
-        if (result !== ProcessNewTransferResultStatusEnum.SUCCESS) {
+          this.heartbeat();
           break;
         }
-      } else {
-        this.logger.log(`There are no success transaction`);
-        const lastTransactionInBatch = transactions.at(-1);
-        if (lastTransactionInBatch) {
-          await this.supportedChainRepository.update(chainId, {
-            latestSolanaTransaction: lastTransactionInBatch.transactionHash,
-            latestBlock: lastTransactionInBatch.slotNumber,
-            lastTxTimestamp: lastTransactionInBatch.transactionTimestamp.toString(),
-            lastTransactionSlotNumber: lastTransactionInBatch.slotNumber,
-          });
-          this.logger.log(
-            `updateSupportedChainBlock; transactionHash: ${lastTransactionInBatch.transactionHash}; slotNumber: ${lastTransactionInBatch.slotNumber};`,
-          );
+        case 'event': {
+          this.heartbeat();
+          // @ts-ignore
+          const event = response.sendEventMessage.event;
+          const nonce = Number(U256Converter.toBigInt(event.submission?.nonce).toString());
+          this.#logger.log(`new event is received nonce: ${nonce}`);
+
+          const submission = this.transformService.generateSubmissionFromSolanaSendEvent(event);
+          this.#submissionsFromSync.push(submission);
+          break;
```

Comment:
Why do you break after first event receiving? Grpc have to return stream with events